lib/utilities/RelayMD5Sum.js

```diff
@@ -0,0 +1,48 @@
+import crypto from 'crypto';
+import { Transform } from 'stream';
+
+/**
+ * This class is designed to compute the md5 hash on multiple streams in relay
+ * race style.
+ */
+export default class RelayMD5Sum extends Transform {
+
+    /**
+     * @constructor
+     * @param {Hash} starterHash - hash from prior stream (if any) to be updated
+     * with new stream here
+     * @param {function} done Callback - This callback is called when
+     * the hash computation is finished, this function need to be synchronous
+     * and being call before the end of the stream object
+     */
+    constructor(starterHash, done) {
```

Comment:
We also don't digest in the relay class so the flush function is different.  Also, the md5sum class is now actually no longer taking a callback but instead emitting an event when the hash is complete since I was getting a race condition for small objects.  To make sure we do not have the same problem here, I am getting rid of the relay flush function altogether and calling the done callback with my new hash before asking for a new chunk.